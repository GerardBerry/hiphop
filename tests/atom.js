"use strict"

var rjs = require("../xml-compiler.js");

var func = function() {
   console.log("atom works!");
}

var prg = <rjs.ReactiveMachine>
  <rjs.loop>
    <rjs.sequence>
      <rjs.pause/>
      <rjs.atom func=${func}/>
    </rjs.sequence>
  </rjs.loop>
</rjs.ReactiveMachine>;

for (var i = 0; i < 15; i++)
   prg.react();
