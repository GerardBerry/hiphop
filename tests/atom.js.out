---- reaction  GO:true RES:true INIT ----
---- SEL:false K0:false K1:true ----

---- reaction  GO:false RES:true ----
atom works!
---- SEL:false K0:false K1:true ----

---- reaction  GO:false RES:true ----
atom works!
---- SEL:false K0:false K1:true ----

---- reaction  GO:false RES:true ----
atom works!
---- SEL:false K0:false K1:true ----

---- reaction  GO:false RES:true ----
atom works!
---- SEL:false K0:false K1:true ----

---- reaction  GO:false RES:true ----
atom works!
---- SEL:false K0:false K1:true ----

---- reaction  GO:false RES:true ----
atom works!
---- SEL:false K0:false K1:true ----

---- reaction  GO:false RES:true ----
atom works!
---- SEL:false K0:false K1:true ----

---- reaction  GO:false RES:true ----
atom works!
---- SEL:false K0:false K1:true ----

---- reaction  GO:false RES:true ----
atom works!
---- SEL:false K0:false K1:true ----

---- reaction  GO:false RES:true ----
atom works!
---- SEL:false K0:false K1:true ----

---- reaction  GO:false RES:true ----
atom works!
---- SEL:false K0:false K1:true ----

---- reaction  GO:false RES:true ----
atom works!
---- SEL:false K0:false K1:true ----

---- reaction  GO:false RES:true ----
atom works!
---- SEL:false K0:false K1:true ----

---- reaction  GO:false RES:true ----
atom works!
---- SEL:false K0:false K1:true ----

